internal/pkg/manifest/svc.go

```diff
@@ -272,17 +273,49 @@ func (a *AdvancedCount) validScalingFields() []string {
 	}
 }
 
-func (a *AdvancedCount) hasScalingFieldsSet() bool {
-	switch a.workloadType {
-	case LoadBalancedWebServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty()
-	case BackendServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty()
-	case WorkerServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.QueueScaling.IsEmpty()
-	default:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty() || !a.QueueScaling.IsEmpty()
+func (a *AdvancedCount) getValidScalingFieldsSet() ([]string, error) {
+	scalingFieldNames := []string{"CPU", "Memory", "Requests", "ResponseTime", "QueueScaling"}
+	validFieldsSet := []string{}
+	invalidFieldsSet := []string{}
+
+	validTags := a.validScalingFields()
+	typ := reflect.ValueOf(*a).Type()
+
+	for _, name := range scalingFieldNames {
+		currStructField, ok := typ.FieldByName(name)
+		if ok {
+			zero := reflect.Zero(currStructField.Type).Interface()
+			value := reflect.ValueOf(a).Elem().FieldByName(name).Interface()
+			if zero != value {
+				tag := currStructField.Tag.Get("yaml")
+				isValid := false
+
+				for _, v := range validTags {
+					if tag == v {
+						validFieldsSet = append(validFieldsSet, tag)
+						isValid = true
+					}
+				}
+				if !isValid {
+					invalidFieldsSet = append(invalidFieldsSet, tag)
+				}
+			}
+		}
+	}
+
+	if len(invalidFieldsSet) > 0 {
+		return validFieldsSet,
```

Comment:
It is a little weird for the return value to be non nil when an error is thrown. Maybe return nil?